src/App.js

```diff
@@ -14,11 +17,93 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+export class App extends React.Component {
+  state = {
+    goods: goodsFromServer,
+    isVisible: false,
+    sortByAlphabet: true,
+    sortByLength: true,
+  }
+
+  sortByLength = () => {
+    this.setState(state => ({
+      goods: [...state.goods].sort((firstGoods, seconGoods) => (
+        state.sortByLength
+          ? firstGoods.length - seconGoods.length
+          : seconGoods.length - firstGoods.length
+      )),
+
+      sortByLength: !state.sortByLength,
+    }));
+  }
+
+  sortByAlphabet = () => {
+    this.setState(state => ({
+      goods: [...state.goods].sort((firstGoods, seconGoods) => (
+        state.sortByAlphabet
+          ? firstGoods.localeCompare(seconGoods)
+          : seconGoods.localeCompare(firstGoods)
+      )),
+
+      sortByAlphabet: !state.sortByAlphabet,
+    }));
+  }
+
+  reverse = () => {
+    this.setState(prevState => ({
+      goods: [...prevState.goods.reverse()],
```

Comment:
спочатку робимо копію, а потім reverse для копії, зараз метод reverse мутує масив, а потім по зміненому масиву створюється копія
```suggestion
      goods: [...prevState.goods].reverse(),
```